pkg/controller/account/account_controller.go

```diff
@@ -528,6 +528,33 @@ func (r *ReconcileAccount) nonCCSAssignAccountID(reqLogger logr.Logger, currentA
 		return err
 	}
 
+	configMap, err := controllerutils.GetOperatorConfigMap(r.Client)
+	if err != nil {
+		reqLogger.Error(err, "failed getting shard name from configmap data")
+		return err
+	}
+
+	hiveName, ok := configMap.Data["shard-name"]
+
+	if ok {
```

Comment:
With the `if/else` thing we talked about when we paired up last week, let's try to fail fast here and check for 

```
if !ok { 
  return fmt.Errorf("shard-name key not available in configmap")
} 
... rest of inputTag operation
```

It's easier to read the code when your errors are caught earlier and returned before the "happy-path" code is executed.